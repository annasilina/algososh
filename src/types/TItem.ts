import {ElementStates} from "./element-states";

export type TItem<T> = {
	value: T;
	state: ElementStates;
}