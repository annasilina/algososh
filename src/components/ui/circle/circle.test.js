import renderer from 'react-test-renderer';
import React from 'react';

import {Circle} from './circle';
import {ElementStates} from '../../../types/element-states';

describe('<Circle />', () => {
	it('should render correctly without letters', () => {
		const element = renderer
			.create(<Circle/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with letters "test"', () => {
		const element = renderer
			.create(<Circle letter='test'/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with value "head" at prop head', () => {
		const element = renderer
			.create(<Circle head='head'/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with ReactElement <Circle /> at prop head ', () => {
		const element = renderer
			.create(<Circle head={<Circle/>}/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with value "tail" at prop tail', () => {
		const element = renderer
			.create(<Circle tail='tail'/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with ReactElement <Circle /> at prop tail ', () => {
		const element = renderer
			.create(<Circle tail={<Circle/>}/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with value 1 at prop index', () => {
		const element = renderer
			.create(<Circle index={1}/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with enabled prop isSmall', () => {
		const element = renderer
			.create(<Circle isSmall/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with default state', () => {
		const element = renderer
			.create(<Circle state={ElementStates.Default}/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with changing state', () => {
		const element = renderer
			.create(<Circle state={ElementStates.Changing}/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});

	it('should render correctly with modified state', () => {
		const element = renderer
			.create(<Circle state={ElementStates.Changing}/>)
			.toJSON();
		expect(element).toMatchSnapshot();
	});
})