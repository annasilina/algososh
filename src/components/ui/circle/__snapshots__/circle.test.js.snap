// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Circle /> should render correctly with ReactElement <Circle /> at prop head  1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> should render correctly with ReactElement <Circle /> at prop tail  1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`<Circle /> should render correctly with changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> should render correctly with default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> should render correctly with enabled prop isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> should render correctly with letters "test" 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> should render correctly with modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> should render correctly with value "head" at prop head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`<Circle /> should render correctly with value "tail" at prop tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`<Circle /> should render correctly with value 1 at prop index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`<Circle /> should render correctly without letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
